### PR TITLE
refactor: 💡 Migrate `worker` detail view table to `Hds::Table`

### PR DESCRIPTION
**Pull request by dvonanderson**
✅ Closes: https://hashicorp.atlassian.net/browse/ICU-11720

:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/JIRA_TICKET_NUMBER)

## Description
- Migrate `worker` detail view for `config_tags` table to `Hds::Table`
- Couldn't use `Form::Field::ListWrapper` due to this view being a `readonly` for the `config_tags`.  I hadto use a normal `Hds::Table` with the `each-in` loop to display the key-value pairs.  
- Also extended the `list-wrapper-field` CSS class into a new class called `config-tags`.  The `list-wrapper-field` styling was perfect to remove the border and some of the padding that comes with the `Hds::Table`, but it was also setup to provide additional styling for the actions in a row in which we don't have.  

### Before:
---
Screenshot 2024-01-09 at 12 58 06

### After:
---
Screenshot 2024-01-09 at 12 58 34

## How to Test
- Click on Vercel link for Admin UI or pull down PR and run it locally using `yarn start`
- Click on "Workers" in the sidebar navigation.
- Click on any highlighted workers in the list and view the "Workers Tags" section
- 

:technologist: [Admin preview](PATH_TO_FEATURE)

## Checklist:

- [x] I have added before and after screenshots for UI changes
~~- [ ] I have added JSON response output for API changes~~
~~- [ ] I have added steps to reproduce and test for bug fixes in the description~~
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~

### CODE REVIEW

**Review by lisbet-alvarez — APPROVED** [review]
LGTM

File: ui/admin/app/styles/app.scss [hunk 1]
```diff
@@ -735,6 +735,19 @@
   /* stylelint-enable selector-class-pattern */
 }
 
+.config-tags {
+  /* stylelint-disable selector-class-pattern */
+  .hds-table__tbody {
+    .hds-table__td {
+      &:last-child {
+        // remove width and white-space styles
+        width: unset !important;
```
Comment (ZedLi) [inline ui/admin/app/styles/app.scss:744]: Is this `!important` needed? Really should avoid any usage of it and if we're using the hds classes (i.e. `hds-table__td` and `hds-table__tbody` we shouldn't be needing it unless the specificity is wrong.

File: ui/admin/app/styles/app.scss [hunk 2]
```diff
@@ -735,6 +735,19 @@
   /* stylelint-enable selector-class-pattern */
 }
 
+.config-tags {
+  /* stylelint-disable selector-class-pattern */
+  .hds-table__tbody {
+    .hds-table__td {
+      &:last-child {
```
Comment (ZedLi) [inline ui/admin/app/styles/app.scss:742]: This could have just been `.hds-table__td:last-child`

**Review by ZedLi — CHANGES_REQUESTED** [review]

File: ui/admin/app/styles/app.scss (hunk 1 above)
Comment (dvonanderson) [reply ui/admin/app/styles/app.scss:744]: @ZedLi thats a great point, I think originally I had the specificity wrong and needed it.  Change incoming.

File: ui/admin/app/styles/app.scss (hunk 2 above)
Comment (dvonanderson) [reply ui/admin/app/styles/app.scss:742]: @ZedLi yes, thank you for the catch

File: ui/admin/app/styles/app.scss [hunk 3]
```diff
@@ -735,6 +735,17 @@
...
+.config-tags {
+  /* stylelint-disable selector-class-pattern */
+  .hds-table__tbody {
+    .hds-table__td:last-child {
+      // remove width and white-space styles
+      width: unset;
+    }
+  }
+}
+/* stylelint-enable selector-class-pattern */
```
Comment (ZedLi) [inline ui/admin/app/styles/app.scss:747]: Wait is this to unset our _own_ styles? We should nest this under `.list-wrapper-field` if so. This should be something like `&.config-tags`

File: ui/admin/app/styles/app.scss [hunk 4]
```diff
@@ -735,6 +735,17 @@
   /* stylelint-enable selector-class-pattern */
 }
 
+.config-tags {
+  /* stylelint-disable selector-class-pattern */
+  .hds-table__tbody {
+    .hds-table__td:last-child {
+      // remove width and white-space styles
```
Comment (ZedLi) [inline ui/admin/app/styles/app.scss:742]: Is this removing white space styles?

File: ui/admin/app/styles/app.scss (hunk 3 above)
Comment (dvonanderson) [reply ui/admin/app/styles/app.scss:747]: @ZedLi got ya I see your point

File: ui/admin/app/styles/app.scss (hunk 4 above)
Comment (dvonanderson) [reply ui/admin/app/styles/app.scss:742]: @ZedLi no just the width, changing comment

**Review by cameronperera — APPROVED** [review]
LGTM

**Review by ZedLi — APPROVED** [review]

**Pull request merged**